mesheryctl/internal/cli/root/pattern/delete.go

```diff
@@ -48,97 +49,154 @@ mesheryctl pattern delete [file | URL]
 			return nil
 		}
 
-		pattern := ""
-		isID := false
-		if len(args) > 0 {
-			pattern, isID, err = utils.ValidId(mctlCfg.GetBaseMesheryURL(), args[0], "pattern")
-			if err != nil {
-				utils.Log.Error(ErrPatternInvalidNameOrID(err))
-				return nil
-			}
-		}
-
-		// Delete the pattern using the id
-		if isID {
-			err := utils.DeleteConfiguration(mctlCfg.GetBaseMesheryURL(), pattern, "pattern")
-			if err != nil {
-				utils.Log.Error(err)
-				return errors.Wrap(err, utils.PatternError(fmt.Sprintf("failed to delete pattern %s", args[0])))
-			}
-			utils.Log.Info("Pattern ", args[0], " deleted successfully")
+		contextID, err := getContextID(mctlCfg)
+		if err != nil {
+			utils.Log.Error(err)
 			return nil
 		}
+
 		deployURL := mctlCfg.GetBaseMesheryURL() + "/api/pattern/deploy"
 		patternURL := mctlCfg.GetBaseMesheryURL() + "/api/pattern"
-
-		// If file path not a valid URL, treat it like a local file path
-		if !govalidator.IsURL(file) {
-			content, err := os.ReadFile(file)
-			if err != nil {
-				utils.Log.Error(utils.ErrFileRead(errors.New(utils.PatternError(fmt.Sprintf("failed to read file %s. Ensure the filename or URL is valid", file)))))
-				return utils.ErrFileRead(errors.New(utils.PatternError(fmt.Sprintf("failed to read file %s. Ensure the filename or URL is valid", file))))
-			}
-
-			patternFile = string(content)
-		} else {
-			// Else treat it like a URL
-			url, path, err := utils.ParseURLGithub(file)
-			if err != nil {
-				utils.Log.Error(utils.ErrParseGithubFile(err, file))
-				return utils.ErrParseGithubFile(err, file)
-			}
-
-			utils.Log.Debug(url)
-			utils.Log.Debug(path)
-
-			var jsonValues []byte
-
-			// Send the URL and path to the server and let it fetch the patternfile and delete
-			// the components
-			if path != "" {
-				jsonValues, _ = json.Marshal(map[string]interface{}{
-					"url":  url,
-					"path": path,
-					"save": false,
-				})
+		// If a file path or URL is provided, delete the pattern using the file
+		if file != "" {
```

Comment:
Could we split delete by file and by  name in 2 methods to make it more readable and easier to follow.

Reply:
Done